core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1876,6 +1877,41 @@ public FileInfo createFile(AlluxioURI path, CreateFileContext context)
     }
   }
 
+  /**
+   * @param rpcContext the rpc context
+   * @param inodePath the path to be created
+   * @param context the method context
+   */
+  private void overwriteFileInternal(RpcContext rpcContext, LockedInodePath inodePath,
+                                     CreateFileContext context)
+      throws FileDoesNotExistException, IOException, InvalidPathException,
+      FileAlreadyExistsException {
+    if (inodePath.fullPathExists()) {
+      Inode currentInode = inodePath.getInode();
+      if (!context.getOptions().getOverwrite()) {
+        throw new FileAlreadyExistsException(
+            "Not allowed to create() (overwrite=false) for existing Alluxio path: "
+                + inodePath.getUri());
+      }
+      // if the fullpath is a file and the option is to overwrite, delete it
+      if (currentInode instanceof InodeFile) {
```

Comment:
can you make the logic here match the original logic in `AbstractFileSystem`?

Reply:
done